dev-lang/vala/vala38-0.38.8.recipe

```diff
@@ -0,0 +1,131 @@
+CHECKSUM_SHA256="2fa746b51cd66e43577d1da06a80b708c2875cadaafee77e9700ea35cf23882c"
+SOURCE_DIR="vala-$portVersion"
+PATCHES="vala38-$portVersion.patchset"
+
+libVersion="0.0.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+PROVIDES="
+	vala$secondaryArchSuffix = $portVersion compat >= 0
+	cmd:vala$secondaryArchSuffix
+	cmd:valac$secondaryArchSuffix
+	cmd:valadoc$secondaryArchSuffix
+	cmd:vapicheck$secondaryArchSuffix
+	cmd:vapigen$secondaryArchSuffix
+	cmd:vala_gen_introspect$secondaryArchSuffix
+	cmd:vala_${portVersion%.*}$secondaryArchSuffix
+	cmd:vala_gen_introspect_${portVersion%.*}$secondaryArchSuffix
+	cmd:valac_${portVersion%.*}$secondaryArchSuffix
+	cmd:valadoc_${portVersion%.*}$secondaryArchSuffix
+	cmd:vapicheck_${portVersion%.*}$secondaryArchSuffix
+	cmd:vapigen_${portVersion%.*}$secondaryArchSuffix
+	lib:libvala_${portVersion%.*}$secondaryArchSuffix = $libVersionCompat
+	lib:libvaladoc_${portVersion%.*}$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libdbus_1$secondaryArchSuffix
+	lib:libffi$secondaryArchSuffix
+	lib:libglib_2.0$secondaryArchSuffix
+	lib:libgvc$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	"
+
+REPLACES="
+	vala${secondaryArchSuffix}_common
+	"
+
+PROVIDES_devel="
+	vala${secondaryArchSuffix}_devel = $portVersion compat >= 0
+	"
+REQUIRES_devel="
+	vala$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libcgraph$secondaryArchSuffix
+	devel:libdbus_1$secondaryArchSuffix
+	devel:libglib_2.0$secondaryArchSuffix
+	devel:libgvc$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf
+	cmd:automake
+	cmd:autoreconf
+	cmd:awk
+	cmd:bison
+	cmd:cmp
+	cmd:diff
+	cmd:flex
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage vala$secondaryArchSuffix \
+	$libDir/libvala-${portVersion%.*}.so.$libVersion \
+	$libDir/libvaladoc-${portVersion%.*}.so.$libVersion \
+	$binDir/vala \
+	$binDir/valac \
+	$binDir/valadoc \
+	$binDir/vapicheck \
+	$binDir/vapigen \
+	$binDir/vala-${portVersion%.*} \
+	$binDir/valac-${portVersion%.*} \
+	$binDir/valadoc-${portVersion%.*} \
+	$binDir/vapicheck-${portVersion%.*} \
+	$binDir/vapigen-${portVersion%.*}
+
+PATCH()
+{
+		sed -i -e 's/-lm/-lroot/' tests/testrunner.sh
+}
+
+BUILD()
+{
+	autoreconf -fi
+	runConfigure ./configure
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	rm -f $libDir/*.la
+	rm -f $libDir/*/*.la
+	rm -f $libDir/*/*/*/*.la
+
+	fixPkgconfig
+
+	#devel and common packages
+	packageEntries devel \
+		$developDir
+}
+
+TEST()
+{
+	#tests run ok up to
+	#runtime_loader: Cannot open file libvaladoc-0.38.so.0: No such file or directory
```

Comment:
I haven't checked this yet, but I'm quite sure it would work with something like:
```diff
 TEST()
 {
-	#tests run ok up to
-	#runtime_loader: Cannot open file libvaladoc-0.38.so.0: No such file or directory
+	LIBRARY_PATH="$sourceDir/.libs${LIBRARY_PATH:+:$LIBRARY_PATH}" \
 	make check
 }
```
Note: This assumes that the share lib is built in a `.libs` subdirectory of the build directory, so you might need to replace `.libs` by something else if that's not the case.

Remark: With the complicated line I'm suggesting, the body of `TEST()` is equivalent to:
* `LIBRARY_PATH="$sourceDir/.libs" make check` if `LIBRARY_PATH` is an empty string, and to
* `LIBRARY_PATH="$sourceDir/.libs:$LIBRARY_PATH" make check` if `LIBRARY_PATH` is not an empty string.

(But it is better to use the complicated line because it handles both cases.)

Reply:
Shared libs seem to be installed at "install" time, so not present in the build section .. ?

Reply:
I haven't tested the recipe yet, but normally, after the call to `make` in `BUILD()`, you should have everything, including the 2 shared libs:
* libvala-0.38.so.0.38.8
* libvaladoc-0.38.so.0.38.8

They might be in a hidden directory (e.g. `.libs`), so the best way to find them is:
```
cd work-0.38.8/sources/vala-0.38.8
find . -name "libvala*.so.*"
```
And `INSTALL()` will not remove them from the build directory, so they will be available to `TEST()` if you use the right directory for the definition of `LIBRARY_PATH`.

Reply:
find . -name "libvala*.so.* doesn't show anything in the $sourceDir ...

Reply:
Well, that's strange, because with vala-0.36.12 (from your other PR, #1920, which I have already tested) I did find a shared lib, `libvala-0.36.so.0.0.0`, in `work-0.36.12/sources/vala-0.36.12/codegen/.libs/` iirc.

Do you get a `work-0.38.8/sources/vala-0.38.8/codegen/.libs/` directory with this PR?

Reply:
Required lib seems to be in /boot/home/haikuports/dev-lang/vala/work-x86-0.38.8/sources/vala-0.38.8/libvaladoc/.libs (didn't uncheck hide dot files), but even then find didn't seem to catch it ...

Reply:
I think you forgot to protect the `libvala*.so*` parameter in your find command.
(find will **_never_** miss a file if it exists, unless your command is not correct, or your filesystem is corrupt :)

Anyway, this should allow `make check` to find the `libvaladoc.so`:
```
TEST()
{
	LIBRARY_PATH="$sourceDir/libvaladoc/.libs${LIBRARY_PATH:+:$LIBRARY_PATH}" \
	make check
}
```